test/test.dart

```diff
@@ -92,9 +95,11 @@ void xtest(String? name, dynamic Function() testFunction) {
   testing.test(name, testFunction, skip: "Test is disabled");
 }
 
-void setupTests(List<String>? args) {
+Future<void> setupTests(List<String>? args) async {
```

Comment:
I think it will be good to separate local tests from sync tests, as it is in .Net. Maybe having another setupSyncTests method would be better. And then to have another sync_test.dart file using this new test setup. So that the old tests will still be working without BaaS.

Reply:
The old tests should still work without baas. The only change I made to `setupTests` is to call `setupBaas`, which is a no-op if the `BAAS_URL` environment variable is not set:

https://github.com/realm/realm-dart/blob/2c43087906deef2c1d25aa6546afc329b70271ae/test/test.dart#L174-L176

Reply:
OK, thanks! Still wondering, what will happen if baasUrl is set but it is wrong or not available?

Reply:
The tests will fail, which I think is the desired behavior - if you are relying on testing some sync functionality (which you probably are if you've set the `BAAS_URL` env variable), then you probably want to know straight away that something is wrong.